misskey-api/src/model/page.rs

```diff
@@ -1,12 +1,136 @@
-use crate::model::id::Id;
+use std::fmt::{self, Display};
 
+use crate::model::{drive::DriveFile, id::Id, user::User};
+
+use chrono::{DateTime, Utc};
 use serde::{Deserialize, Serialize};
+use thiserror::Error;
 
 #[derive(Serialize, Deserialize, Debug, Clone)]
+#[serde(rename_all = "camelCase")]
 pub struct Page {
     pub id: Id<Page>,
-    #[serde(flatten)]
-    pub content: serde_json::Map<String, serde_json::Value>,
+    pub created_at: DateTime<Utc>,
+    pub updated_at: DateTime<Utc>,
+    pub user_id: Id<User>,
+    pub user: Box<User>,
+    pub content: Content,
+    pub variables: Variables,
+    pub title: String,
+    pub name: String,
+    #[serde(default)]
+    pub summary: Option<String>,
+    pub align_center: bool,
+    pub hide_title_when_pinned: bool,
+    pub font: Font,
+    #[cfg(feature = "12-31-0")]
+    #[cfg_attr(docsrs, doc(cfg(feature = "12-31-0")))]
+    pub script: String,
+    pub eye_catching_image_id: Option<Id<DriveFile>>,
+    pub eye_catching_image: Option<Box<DriveFile>>,
+    pub attached_files: Vec<DriveFile>,
+    pub liked_count: u64,
+    #[serde(default)]
+    pub is_liked: Option<bool>,
 }
 
 impl_entity!(Page);
+
+#[derive(Serialize, Deserialize, Debug, Clone)]
+#[serde(rename_all = "camelCase")]
+pub struct PageLike {
+    pub id: Id<PageLike>,
+    pub page: Page,
+}
+
+impl_entity!(PageLike);
+
+#[derive(Serialize, Deserialize, Default, Debug, Clone)]
+pub struct Content(pub Vec<serde_json::Map<String, serde_json::Value>>);
+
+#[derive(Debug, Error)]
+#[error("invalid content: {0}")]
+pub struct ParseContentError(#[from] serde_json::Error);
+
+impl std::str::FromStr for Content {
+    type Err = ParseContentError;
+
+    fn from_str(s: &str) -> Result<Content, Self::Err> {
+        serde_json::from_str(s).map_err(Into::into)
+    }
+}
+
+impl TryFrom<serde_json::Value> for Content {
+    type Error = ParseContentError;
```

Comment:
I think `serde_json::Error` is better here (`TryFrom` doesn't do `Parse`)

```suggestion
    type Error = serde_json::Error;
```

same applies to `Variables`.